src/zope/app/testing/dochttp.py

```diff
@@ -79,31 +80,34 @@ def dochttp(args=sys.argv[1:], default=None):
     names = [name[:-len(".request")]
              for name in os.listdir(directory)
              if name.startswith(options.prefix) and name.endswith('.request')
-             ]
+    ]
     names.sort()
 
-    extre = re.compile("[.](\w+)$")
+    extre = re.compile(r"[.](\w+)$")
 
     for name in names:
-        requests =  Requests(
-                        open(os.path.join(directory, name + ".request"), 'rb'),
-                        options.skip_request_header,
-                        )
-        responses = Responses(
-                        open(os.path.join(directory, name + ".response"), 'rb'),
-                        options.skip_response_header,
-                        )
-
-        # We use map so as *not* to truncate at shortest input.
-        # We want an error if the number of requests and responses
-        # is different.
-        for request, response in map(None, requests, responses):
+        with open(os.path.join(directory, name + ".request"), 'rb') as f:
+            requests = list(Requests(
+                f,
+                options.skip_request_header,
+            ))
+        with open(os.path.join(directory, name + ".response"), 'rb') as f:
+            responses = list(Responses(
+                f,
+                options.skip_response_header,
+            ))
+
+        if len(requests) != len(responses):
+            raise ValueError("Expected equal length requests and responses")
```

Comment:
This is a command-line tool.  If it finds a pair of files on disk that do not pass some requirement (such as having matching numbers of data records), the right thing to do is to print an error message (that mentions names of the files involved!) and exit with a non-zero exit status.  So I suggest something like

```python
sys.exit("Expected equal numbers of requests and responses in %s"
         % os.path.join(directory, name + ".*"))
```